clang/lib/CIR/CodeGen/CIRGenDecl.cpp

```diff
@@ -94,10 +203,59 @@ void CIRGenFunction::emitVarDecl(const VarDecl &d) {
 
   assert(d.hasLocalStorage());
 
-  assert(!cir::MissingFeatures::opAllocaVarDeclContext());
+  CIRGenFunction::VarDeclContext varDeclCtx{*this, &d};
   return emitAutoVarDecl(d);
 }
 
+void CIRGenFunction::emitScalarInit(const Expr *init, mlir::Location loc,
+                                    LValue lvalue, bool capturedByInit) {
+  Qualifiers::ObjCLifetime lifetime = Qualifiers::ObjCLifetime::OCL_None;
```

Comment:
Why not:
```
  Qualifiers::ObjCLifetime lifetime = lvalue.getObjCLifetime();
```
as in main codegen?

Reply:
That's not implemented yet. CIR has its own LValue class, and getObjCLifetime() hasn't been added there.